src/sentry/middleware/integrations/parsers/discord.py

```diff
@@ -42,6 +49,17 @@ def discord_request(self) -> DiscordRequest | None:
         self._discord_request: DiscordRequest = self.view_class.discord_request_class(drf_request)
         return self._discord_request
 
+    def get_async_region_response(self, regions: List[Region]) -> HttpResponse:
+        webhook_payload = ControlOutbox.get_webhook_payload_from_request(request=self.request)
+        convert_to_async_discord_response.apply_async(
+            kwargs={
+                "payload": dataclasses.asdict(webhook_payload),
+                "region_names": [r.name for r in regions],
+            }
+        )
```

Comment:
If this job is processed before discord gets the initial reply will it 'work'? We've had problems in the past with celery jobs being picked up before database transactions complete, and this looks like it could have a similar timing race.

I don't see a good way to ensure the task is enqueued after the response has been delivered though. You could override `Response.close` but that seems hacky, the [request_finished](https://docs.djangoproject.com/en/3.2/ref/signals/#request-finished) could work but attaching the signal handler and detaching it will be a bit clunky.

Reply:
I can take a look at this, but you're right in assuming it'd break. Discord needs the response first prior to accepting the hooks.

Reply:
from a separate discussion: won't be addressing immediately, but i added a doc string to mention that this race _could_ happen